python/test/unit/language/test_core.py

```diff
@@ -131,7 +132,7 @@ def check_type_supported(dtype, device):
         cc = torch.cuda.get_device_capability()
         if cc[0] < 8 and (dtype is tl.bfloat16 or dtype == "bfloat16" or dtype is torch.bfloat16):
             pytest.skip("bfloat16 is only supported on NVGPU with cc >= 80")
-        if cc[0] < 9 and (dtype is tl.float8e4nv or dtype == "float8e4nv"):
+        if cc[0] < 9 and dtype in {tl.float8e4nv, "float8e4nv", "float8_e4m3fn"}:
```

Comment:
Actually we want to support e4m3 float8 format on pre sm_90 targets, that is why we have the conversion in inline ptx. We should not skip it and make sure the lowering still supports it. (I don't think your code break it?)

Reply:
to be clear, pre-sm90 targets support float8e4b15 for now, which is not quite the same as float8e4nv since it doesn't handle nans and has a different exponent bias